### PR TITLE
Raise error when each channel only has 1 value in batch norm

### LINKED ISSUE #1381: BatchNorm shouldn't use Bessel's correction for a batch size of 1
Right now it makes everything `NaN`.

### PR DESCRIPTION
**Pull request by ssnl**
Fixes #1381 by raising error when a channel observes only 1 value for a batch in batch norm.

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by colesbury** [comment]
This check isn't right. For example, we allow 1-element per channel in training mode is False.

**Comment by SsnL** [comment]
@colesbury good point, I'll write a PR to fix that case. Any other case this may be wrong?
